### DISCUSSION TITLE
how to add a survey

### DISCUSSION THREAD

**Discussion by jnash190** [tatool/tatool-web#148] [Q&A] [answered]
Hello, I am new to tatool and I am trying to add some survey questions after completing the cognitive tasks (multiple choice questions, likert scale, select all that apply). Does anyone know how to add a component that resembles this?

**Comment by cvonbastian** [comment, marked as answer]
Hi Jenna, and welcome to the Tatool Web community! 

There are basically 2 options. The first is to **program an executable** that can do this. We have a basic survey executable using [SurveyJS](https://surveyjs.io/) that can do drop-downs and radio buttons. I'd be happy to share that with you if you'd like to build on that. Second - and much easier - you could use any existing survey tool (e.g., Qualtrics if your institution has a subscription) and use the the new "**Redirect URL**" in the module editor to link to your survey. Once participants have completed your cognitive tasks in Tatool Web, they will then be automatically redirected to your survey. You can of course also do the opposite and start with the survey and then put a link to your Tatool Web module at the end of it. 

Best, Claudia

**Reply by jnash190** [reply]
Hi Claudia! Thank you for your help! 

I was wondering where I go to access the redirect URL module. Is it part of a specific executable?

**Reply by cvonbastian** [reply]
It's one of the properties you can set in the general settings of any module, see below screenshot: 
![Screenshot 2021-02-24 16 30 23](https://user-images.githubusercontent.com/9940752/109033178-46e1d080-76be-11eb-9f9c-e452fd0a0e61.png)
You can just enter the URL in that textbox.

**Reply by jnash190** [reply]
Hello, do you know how we can match the qualtrics responses to the tatool responses?

**Reply by jnash190** [reply]
Any assistance on this issue would be much appreciated!! Thank you again for all your help

**Reply by alocher** [reply]
Hey, we've got a similar question which I've just answered [here](https://github.com/tatool/tatool-web/discussions/151#discussioncomment-485292).

**Reply by abiUni** [reply]
Hi @cvonbastian , hi @jnash190,
thank you for a great tool making so many scientists' lives easier!!
I was wondering if you could share with me the basic survey executable you mentioned in this post, as I would like to reduce hops between tools/tabs in my study.
Thank you!

**Reply by cvonbastian** [reply]
Hi @abiUni! It's great to hear that you find Tatool Web useful, thank you! 
I've uploaded the basic executable tatoolSurvey to the tatool-share project, you can find it here: https://github.com/tatool/tatool-web/tree/master/app/projects/public/tatool-share/executables
I've added a few example questions just so you could run this executable in a module (add executable to a new module, choose tatool-share project, then enter tatoolSurvey as name). The questions are hardcoded in the executable, so it will only need a single iteration. 
To create your own questions, you can head to https://surveyjs.io/create-survey and then copy & paste the code from the JSON editor tab. It's actually quite easy even if it's not particularly elegant to just copy/paste the code. :) 
I hope that helps, and good luck with your study!
Claudia

**Reply by abiUni** [reply]
Hi @cvonbastian,
thank you very much! that's exactly what I was looking for :) 
Abi
